app/modules/wps.c

```diff
@@ -0,0 +1,73 @@
+// Module for WPS
+// by F.J. Exoo
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+
+static int wps_callback_ref;
+
+// Lua: wps.disable()
+static int ICACHE_FLASH_ATTR wps_disable(lua_State* L)
+{
+  wifi_wps_disable();
+  return 0;
+}
+
+// Lua: wps.enable()
+static int ICACHE_FLASH_ATTR wps_enable(lua_State* L)
+{
+  wifi_wps_enable(WPS_TYPE_PBC);
+  return 0;
+}
+
+// WPS start callback function
+LOCAL void ICACHE_FLASH_ATTR user_wps_status_cb(int status)
+{
+  lua_State *L = lua_getstate();
+
+  if (wps_callback_ref != LUA_NOREF) {
+    lua_rawgeti(L, LUA_REGISTRYINDEX, wps_callback_ref);
+    lua_pushinteger(L, status);
+    lua_call(L, 1, 0);
+  }
+}
+
+// Lua: wps.start( function())
+static int ICACHE_FLASH_ATTR wps_start(lua_State* L)
+{
+  // retrieve callback arg (optional)
+  luaL_unref(L, LUA_REGISTRYINDEX, wps_callback_ref);
+
+  wps_callback_ref = LUA_NOREF;
+
+  if (lua_type(L, 1) == LUA_TFUNCTION || lua_type(L, 1) == LUA_TLIGHTFUNCTION)
+    wps_callback_ref = luaL_ref(L, LUA_REGISTRYINDEX);
```

Comment:
There will be issue when more than one argument is passed, so call `lua_pushvalue(L, 1)` before `luaL_ref`